staging/src/k8s.io/apimachinery/pkg/util/diff/compare.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2020 Authors of Arktos.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diff
+
+import "time"
+
+var allowedNTPDiffInMilliSecond = uint64(5000)
+
+// For backwards compatible - can still use ETCD revision # starting from 1 and increases by 1 as backend storage
+var v2MinRevision = getRevisionNumber(time.Date(2020, 5, 10, 0, 0, 0, 0, time.UTC).UnixNano(), 0, 1)
+
+// RevisionIsNewer is used in event comparison to check whether revision 1 is newer than
+// revision 2 and should be sent/accepted/processed
+func RevisionIsNewer(revision1, revision2 uint64) bool {
+	if revision1 < v2MinRevision || revision2 < v2MinRevision {
+		return revision1 > revision2
+	}
+
+	result, isSameCluster := revisionCompare(revision1, revision2)
+	if isSameCluster {
+		return result > 0
+	}
+	// if from different cluster, allow sent result == 0
+	return result >= 0
```

Comment:
Shall we move the logic of isSameCluster into function revisionCompare()?  So this function will be very simple: if old format compare directly, otherwise call new format's comparison. Now the logics of isSameCluster are in two places and hard to understand.

Reply:
RevisionIsNewer() is already very simple. I would like not to over-complicate the logic of revisionCompare().

Reply:
Will keep the original as there is only one public function.